sycl/include/sycl/ext/oneapi/experimental/named_sub_group_sizes.hpp

```diff
@@ -0,0 +1,45 @@
+//== named_sub_group_sizes.hpp --- SYCL extension for named sub-group sizes ==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include <sycl/ext/oneapi/kernel_properties/properties.hpp>
+
+namespace sycl {
+inline namespace _V1 {
+namespace ext::oneapi::experimental {
+
+struct named_sub_group_size {
+  static constexpr uint32_t primary = 0;
+  static constexpr uint32_t automatic = -1;
```

Comment:
I agree, but I thought there was kind of no choice with how the spec for the extension is defined right now. 
My thought process is that since the spec defines the named sub group size properties as values of the `sub_group_size` property, if, for example, we launch a kernel with the property `sub_group_size_automatic`, this is the same as passing it `sub_group_size<automatic>`. so `automatic` (or `primary`) cannot equal the common sub group sizes like `1`, `2`, `4`, etc. without interfering with the regular behavior of the `sub_group_size` property. so I assigned primary to `0` (would not be a sensible sub group size) and automatic to `-1` (as an unsigned integer, would be an unrealisticly sized sub group size).

Or did you just mean using a "negative" value for an unsigned value?

Reply:
I partially misunderstood the extension. I think it's fine, though I would personally prefer an unsigned value for this. Not blocking.